va/validation-authority.go

```diff
@@ -281,7 +270,9 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 				return fmt.Errorf("Invalid port number in redirect")
 			}
 		} else if strings.ToLower(req.URL.Scheme) == "https" {
-			reqPort = "443"
+			reqPort = strconv.Itoa(va.httpsPort)
+		} else {
+			reqPort = strconv.Itoa(va.httpPort)
```

Comment:
Yes. We don't currently do any redirects inside our integration tests, but if we do, we would have them redirect to a port we can bind to in the tests.